parser_library/src/processing/statement_analyzers/hit_count_analyzer.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2023 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+#ifndef PROCESSING_HIT_COUNT_ANALYZER_H
+#define PROCESSING_HIT_COUNT_ANALYZER_H
+
+#include <functional>
+#include <string>
+#include <string_view>
+#include <unordered_map>
+
+#include "context/hlasm_context.h"
+#include "protocol.h"
+#include "statement_analyzer.h"
+#include "utils/general_hashers.h"
+#include "utils/resource_location.h"
+
+namespace hlasm_plugin::parser_library::processing {
+struct hit_count_details
+{
+    range r;
+    size_t count;
+};
+
+struct stmt_hit_count_details
+{
+    std::unordered_map<size_t, hit_count_details> stmt_hc_map;
+    bool is_external_macro = true;
+};
```

Comment:
Does it make sense to have lineno->range map here?
Would perhaps a simple ordered vector of line, count pair be more appropriate structure?
Or maybe just a vector, index == lineno, and the content is count (+ "contains statement" bool).